kubernetes_asyncio/config/kube_config.py

```diff
@@ -201,6 +212,99 @@ def set_active_context(self, context_name=None):
         self.token = "Bearer %s" % config['access-token']
         return self.token
 
+    async def _load_oid_token(self):
+        provider = self._user['auth-provider']
+
+        if 'config' not in provider:
+            raise ValueError('oidc: missing configuration')
+
+        if 'id-token' not in provider['config']:
+            await self._refresh_oidc(provider)
+            self.token = 'Bearer {}'.format(provider['config']['id-token'])
+            return self.token
+
+        parts = provider['config']['id-token'].split('.')
+
+        if len(parts) != 3:
+            raise ValueError('oidc: JWT tokens should contain 3 period-delimited parts')
+
+        id_token = parts[1]
+        # Re-pad the unpadded JWT token
+        id_token += (4 - len(id_token) % 4) * '='
+        jwt_attributes = json.loads(base64.b64decode(id_token).decode('utf8'))
+        expires = jwt_attributes.get('exp')
+
+        if (
+            expires is not None and
+            _is_expired(datetime.datetime.utcfromtimestamp(expires))
+        ):
+            await self._refresh_oidc(provider)
+
+        self.token = 'Bearer {}'.format(provider['config']['id-token'])
+        return self.token
+
+    async def _refresh_oidc(self, provider):
+        if 'refresh-token' not in provider['config']:
+            raise ValueError('oidc: No valid id-token, and cannot refresh without refresh-token')
+
+        with tempfile.NamedTemporaryFile(delete=True) as certfile:
+            configuration = Configuration()
+            cert_auth_data = self._retrieve_oidc_cacert(provider)
+            if cert_auth_data is not None:
+                certfile.write(cert_auth_data)
+                certfile.flush()
+                configuration.ssl_ca_cert = certfile.name
+
+            client = ApiClient(configuration=configuration)
+            issuer_url = provider['config']['idp-issuer-url']
+            response = await client.request(
+                method='GET',
+                url='{}/.well-known/openid-configuration'.format(issuer_url.rstrip("/"))
+            )
+
+            if response.status != 200:
+                raise ValueError('oidc: failed to query metadata endpoint')
+
+            well_known_data = json.loads(response.data)
+
+            self._perform_oauth_refresh(provider, client, well_known_data)
+
+    def _perform_oauth_refresh(self, provider, client, well_known_data):
+        # It's not optimal that this isn't async, but there's no existing well-supported
```

Comment:
Yes, and it was a reason why I didn't copy this from the official library. The main goal of this library is to use only asynchronous calls.